src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/TypeImportCompletionProvider.cs

```diff
@@ -4,45 +4,27 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Completion.Providers;
-using Microsoft.CodeAnalysis.CSharp.Extensions;
-using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
 using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
 {
-    internal class TypeImportCompletionProvider : AbstractTypeImportCompletionProvider
+    internal sealed class TypeImportCompletionProvider : AbstractTypeImportCompletionProvider
     {
         internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
-            => CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+            => ImportCompletionProviderHelper.IsInsertionTrigger(text, characterPosition, options);
```

Comment:
interesting.  why does tihs need a different insertion trigger?

Reply:
Nothing has changed actually, just pulled it into the helper class.

Reply:
i'm confused though. why do you need an IsInsertionTrigger in your helper class?  Why not have this still call directly into CompletionUtilities?  This makes it seem as if there's special logic here.

Reply:
Sure, fixed.